Code/GraphMol/Substruct/SubstructMatch.cpp

```diff
@@ -131,32 +216,46 @@ class MolMatchFinalCheckFunctor {
       mOrder.insert(mOrder.end(), unmatchedNeighbors, -1);
 
       INT_LIST moOrder;
-      ROMol::OEDGE_ITER dbeg, dend;
-      boost::tie(dbeg, dend) = d_mol.getAtomBonds(mAt);
-      while (dbeg != dend) {
-        int dbidx = d_mol[*dbeg]->getIdx();
+      for (const auto &bond : make_iterator_range(d_mol.getAtomBonds(mAt))) {
+        int dbidx = d_mol[bond]->getIdx();
         if (std::find(mOrder.begin(), mOrder.end(), dbidx) != mOrder.end()) {
           moOrder.push_back(dbidx);
         } else {
           moOrder.push_back(-1);
         }
-        ++dbeg;
       }
 
       int mPermCount =
           static_cast<int>(countSwapsToInterconvert(moOrder, mOrder));
 
-      bool requireMatch = qPermCount % 2 == mPermCount % 2;
-      bool labelsMatch = qAt->getChiralTag() == mAt->getChiralTag();
+      const bool requireMatch = qPermCount % 2 == mPermCount % 2;
+      const bool labelsMatch = qAt->getChiralTag() == mAt->getChiralTag();
+      const bool matchOK = requireMatch == labelsMatch;
 
-      if (requireMatch != labelsMatch) {
+      // if this is not part of a stereogroup and doesn't match, return false
```

Comment:
I don't think so, because it's OK for matchOK to be false if the atom is in a stereogroup. If it's in a stereogroup, what matters is that all atoms in the stereogroup are the same matchiness.  e.g. `C[C@@H](O)[C@@H](CC)F` and `"C[C@H](O)[C@H](CC)F` both match `C[C@H](O)[C@H](CC)F |&1:1,3|`, but `C[C@H](O)[C@@H](CC)F` doesn't.

Reply:
Right. Got it.